python/kfserving/kfserving/storage.py

```diff
@@ -27,23 +27,25 @@
 
 class Storage(object): # pylint: disable=too-few-public-methods
     @staticmethod
-    def download(uri: str) -> str:
-        logging.info("Copying contents of %s to local", uri)
+    def download(uri: str, out_dir: str = None) -> str:
```

Comment:
ooo can you mkdtemp inline ;)

Reply:
would be a terrible idea... because python...
>Python’s default arguments are evaluated once when the function is defined, not each time the function is called (like it is in say, Ruby). This means that if you use a mutable default argument and mutate it, you will and have mutated that object for all future calls to the function as well.

https://docs.python-guide.org/writing/gotchas/

Reply:
This actually sounds good to me. The outdir is scoped to the lifetime of the program :)